src/lua/luaTask.c

```diff
@@ -57,78 +58,47 @@ static enum {
 
 static lua_State *g_lua;
 static xSemaphoreHandle xLuaLock;
-static unsigned int lastPointer;
 static size_t onTickSleepInterval;
+static size_t lua_mem_size;
 
-#ifdef ALLOC_DEBUG
-static int g_allocDebug = 0;
-#endif
-void * myAlloc (void *ud, void *ptr, size_t osize,size_t nsize)
+static void* myAlloc(void *ud, void *ptr, size_t osize, size_t nsize)
 {
-
-#ifdef ALLOC_DEBUG
-    if (g_allocDebug) {
-        SendString("myAlloc- ptr:");
-        SendUint((unsigned int)ptr);
-        SendString(" osize:");
-        SendInt(osize);
-        SendString(" nsize:");
-        SendInt(nsize);
-    }
-#endif
-
-    if (nsize == 0) {
-        portFree(ptr);
-#ifdef ALLOC_DEBUG
-        if (g_allocDebug) {
-            SendString(" (free)");
-            SendCrlf();
-        }
-#endif
-        return NULL;
-    } else {
-        void *newPtr;
-        if (osize != nsize) {
-            newPtr = portRealloc(ptr, nsize);
-        } else {
-            newPtr = ptr;
+        const int delta = nsize - osize;
+        const size_t new_lua_mem_size = lua_mem_size + delta;
+
+        if (nsize == 0) {
+                pr_debug_int_msg("[lua] RAM Freed: ", abs(delta));
+                portFree(ptr);
+                lua_mem_size = new_lua_mem_size;
+                return NULL;
         }
-#ifdef ALLOC_DEBUG
-        if (g_allocDebug) {
-            if (ptr != newPtr) {
-                SendString(" newPtr:");
-                SendUint((unsigned int)newPtr);
-            } else {
-                SendString(" (same)");
-            }
-            SendCrlf();
+
+        if (LUA_MEM_MAX && LUA_MEM_MAX < new_lua_mem_size) {
+                pr_info("[lua] Memory ceiling hit: ");
```

Comment:
Good. If no other issues then shipit!

Brent Picasso
Autosport Labs
Technology for Race and Street
On Sep 30, 2015 5:17 PM, "Andrew Stiegmann" <notifications@github.com>
wrote:

> In src/lua/luaTask.c
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/91#discussion_r40868581>
> :
>
> >          }
> > -#ifdef ALLOC_DEBUG
> > -        if (g_allocDebug) {
> > -            if (ptr != newPtr) {
> > -                SendString(" newPtr:");
> > -                SendUint((unsigned int)newPtr);
> > -            } else {
> > -                SendString(" (same)");
> > -            }
> > -            SendCrlf();
> > +
> > +        if (LUA_MEM_MAX && LUA_MEM_MAX < new_lua_mem_size) {
> > +                pr_info("[lua] Memory ceiling hit: ");
>
> Left this as an info because it has to happen 3 times before we will reset
> lua. The action that actually resets lua due to memory issues is at the
> warning level. Good?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/91/files#r40868581>
> .
>